dz2.2/dz2.2.ts

```diff
@@ -0,0 +1,389 @@
+
+    private _groups: Group[];
+    private _name: string;
+    private _description: string;
+
+    constructor(groups: Group[], name: string, description: string) {
+        this._groups = groups;
+        this._name = name;
+        this._description = description;
+    }
+
+    public get groups(): Group[] {
+        return this._groups;
+    }
+
+    public set groups(groups: Group[]) {
+        this._groups = groups;
+    }
+
+    public get name(): string {
+        return this._name;
+    }
+
+    public set name(name: string) {
+        this._name = name;
+    }
+
+    public get description(): string {
+        return this._description;
+    }
+
+    public set description(description: string) {
+        this._description = description;
+    }
+
+    public addGroup(group: Group) {
+        this._groups.push(group);
+    }
+
+    public removeGroup(group: Group) {
+        let newGroups: Group[] = [];
+        for (var i = 0; i < this._groups.length; i++) {
+            if (this._groups[i].directionName !== group.directionName && this._groups[i].levelName !== group.levelName) {
+                newGroups.push(this._groups[i]);
+            }
+        }
+
+        this._groups = newGroups;
+        return this._groups.length;
+    }
+}
+
+class Group {
+    // implement getters for fields and 'add/remove student' and 'set status' methods
+
+    private _area: Area;
+    private _status: string;
+    private _students: Student[]; // Modify the array so that it has a valid toSorted method*
+
+    private _directionName: string;
+    private _levelName: string;
+
+    constructor(directionName: string, levelName: string, area: Area, status: string, students: Student[]) {
+        this._directionName = directionName;
+        this._levelName = levelName;
+        this._area = area;
+        this._status = status;
+        this._students = students;
+    }
+
+    public showPerformance(): Student[] {
+        const sortedStudents = this._students.toSorted((a, b) => b.getPerformanceRating() - a.getPerformanceRating());
+        return sortedStudents;
+    }
+
+    public get directionName(): string {
+        return this._directionName;
+    }
+
+    public set directionName(directionName: string) {
+        this._directionName = directionName;
+    }
+
+    public get area(): Area {
+        return this._area;
+    }
+
+    public set area(area: Area) {
+        this._area = area;
+    }
+
+    public get status(): string {
+        return this._status;
+    }
+
+    public set status(status: string) {
+        this._status = status;
+    }
+
+    public get levelName(): string {
+        return this._levelName;
+    }
+
+    public set levelName(levelName: string) {
+        this._levelName = levelName;
+    }
+
+    public get students(): Student[] {
+        return this._students;
+    }
+
+    public set students(students: Student[]) {
+        this._students = students;
+    }
+
+    public addStudent(student: Student) {
```

Comment:
Please, do not forget to explicitly add type annotation for the return function value.

Reply:
Updated.